middleware/proxy_1_11_test.go

```diff
@@ -50,4 +52,33 @@ func TestProxy_1_11(t *testing.T) {
 	e.ServeHTTP(rec, req)
 	assert.Equal(t, "/api/users", req.URL.Path)
 	assert.Equal(t, http.StatusBadGateway, rec.Code)
+
+	// client closed connection
+	HTTPTarget := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		time.Sleep(5 * time.Second)
```

Comment:
Using time.sleep with seconds makes test execution very slow (5sec)

You can replace with test case with this:
```go
func TestClientCancelConnectionResultsHTTPCode499(t *testing.T) {
	timeoutStopChan := make(chan struct{}, 0)
	HTTPTarget := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		<-timeoutStopChan // wait until we have asserted timeout
		w.WriteHeader(http.StatusOK)
	}))
	defer HTTPTarget.Close()
	targetURL, _ := url.Parse(HTTPTarget.URL)
	target := &ProxyTarget{
		Name: "target",
		URL:  targetURL,
	}
	rb := NewRandomBalancer(nil)
	assert.True(t, rb.AddTarget(target))

	e := echo.New()
	e.Use(Proxy(rb))
	rec := httptest.NewRecorder()
	req := httptest.NewRequest(http.MethodGet, "/", nil)
	ctx, cancel := context.WithCancel(req.Context())
	req = req.WithContext(ctx)

	go func() {
		startTime := time.Now()
		time.Sleep(10 * time.Millisecond)
		cancel()
		endTime := time.Now()
		assert.Less(t, endTime.Sub(startTime).Seconds(), float64(5))
	}()

	e.ServeHTTP(rec, req)

	assert.Equal(t, 499, rec.Code)
	timeoutStopChan <- struct{}{} // signal that handler can return and not block ending of this test
}
```

Reply:
@aldas why you're asserting timeout here again?

Reply:
I am suggesting that by changing test case you added the execution time would be faster. Currently that test function completes in 5 seconds to me.
Also it is clearer then different test cases are separated from each other. That is why I pasted example as completely different function.

Reply:
@aldas I've done it in another way. PTAL